newshomepages/utils.py

```diff
@@ -463,3 +466,173 @@ def intcomma(value):
+        "#didomi-host",  # Common cookies popup
+        ".fc-ab-root",  # Ad blocker popup
+        ".fancybox-overlay",  # Popup overlay
+        ".fancybox-overlay-fixed",
+        ".ab_widget_container_popin-image",  # Popover
+        ".ab_widget_container_popin-image_content",
+        ".ab_widget_container_popin-image_overlay",
+        ".fc-ab-root",  # Popover
+        ".modal-backdrop",  # Modal popup
+        "#dth-borderless-modal-alt",
+        "#floatingBlockerGuide",  # Popup overlay
+        ".ctct-popup-overlay",  # Popup overlay
+        ".ctct-popup-wrapper",
+        ".message-container",  # Popup
+        ".teg-container",
+        'div[id^="sp_message_container"]',  # Popover
+        ".wisepops-root",  # Popover
+        "#wisepops-root",
+        ".widget_eu_cookie_law_widget",  # GDPR poup
+        ".fancybox-overlay",  # Ad overlay
+        ".signup-box",  # Popup ad
+        ".newspack-lightbox",  # Popup
+        ".adunit-googleadmanager",  # Google ad
+        ".mc-modal",  # Newsletter popup
+        ".mc-modal-bg",
+        ".blockNavigation",
+        ".ab-iam-root",
+        ".omaha-background",  # Takeover ad
+        ".sqs-popup-overlay",  # Overlay popup
+        "#gcomPromo",  # Overlay found on Brazilian sites
+        ".Campaign",
+    ]
+    target_str = ",".join(target_list)
+    return f"""
+        document.querySelectorAll('{target_str}').forEach(el => el.remove());
+        var styleSheet = document.createElement('style');
+        styleSheet.innerText = '{target_str} {{ display: none !important; }}';
+        document.head.appendChild(styleSheet);
+     """
+
+
+def _load_new_page_disable_javascript(
+    context: BrowserContext,
+    url: str,
+    handle: str,
+    wait_seconds: int = 5,
+    full_page: bool = False,
+):
+    """Load the page with javascript blocking."""
+    # Create an empty tab
+    page = context.new_page()
+    # Open the page
+    print(f"Opening {url}")
+    page.goto(url, timeout=60000)
+
+    # Give it a beat
+    print(f"Waiting {wait_seconds} seconds")
+    time.sleep(wait_seconds)
+    common_javascript = _get_common_blocking_javascript()
+
+    print("Executing common JavaScript")
+    page.evaluate(common_javascript)
+
+    # If there's custom javascript for this site, run it
+    custom_javascript = get_javascript(handle)
+    if custom_javascript:
+        print("Executing custom JavaScript")
+        page.evaluate(custom_javascript)
+
+    # Hide the scrollbars
+    print("Hiding scrollbars with CSS")
+    css = """document.body.style.overflow = 'hidden';"""
+    page.evaluate(css)
+
+    # If we're saving full-page content, we need to scroll to the bottom to make sure all the images get fetched.
+    if full_page:
+        print("scrolling to bottom")
+        scroll_height = page.evaluate("document.body.offsetHeight")
+        for i in range(0, scroll_height + 500, 500):
+            page.evaluate("scroll(0, %s)" % i)
+            time.sleep(1)
```

Comment:
@simonw, we are adding a full page screenshotting option here. To make sure all the lazy loading images come in, @alex2awesome has added this little trick. What do you think? You see any potential bugaboos?

Reply:
i updated this to better handle infinite scroll